pkg/planner/core/index_join_path.go

```diff
@@ -630,6 +632,43 @@ func getIndexJoinIntPKPathInfo(ds *logicalop.DataSource, innerJoinKeys, outerJoi
 	return keyOff2IdxOff, newOuterJoinKeys, ranges, chosenPath, true
 }
 
+// getBestIndexJoinInnerTaskByProp tries to build the best inner child task from ds for index join by the given property.
+func getBestIndexJoinInnerTaskByProp(ds *logicalop.DataSource, prop *property.PhysicalProperty,
+	opt *optimizetrace.PhysicalOptimizeOp, planCounter *base.PlanCounterTp) (base.Task, int64, error) {
+	// the below code is quite similar from the original logic
+	// reason1: we need to leverage original indexPathInfo down related logic to build constant range for index plan.
+	// reason2: the ranges from TS and IS couldn't be directly used to derive the stats' estimation, it's not real.
+	// reason3: skyline pruning should prune the possible index path which could feel the runtime EQ access conditions.
+	cntPlan := int64(0)
```

Comment:
cntPlan is for `nth_plan` hint, but it's not used, I suggest that we can remove this to simplify our code.

Reply:
Make sense